lib/apartment/active_record/postgresql_adapter.rb

```diff
@@ -0,0 +1,16 @@
+module Apartment::PostgreSqlAdapterPatch
+  def default_sequence_name(table, _column)
+    res = super
+    schema_prefix = "#{Apartment::Tenant.current}."
+    if res.starts_with?(schema_prefix) && Apartment.excluded_models.none?{|m| m.constantize.table_name == table}
+      res.delete_prefix!("#{Apartment::Tenant.current}.")
+    end
+    res
+  end
+end
+
+require 'active_record/connection_adapters/postgresql_adapter'
+
+class ActiveRecord::ConnectionAdapters::PostgreSQLAdapter
+  include Apartment::PostgreSqlAdapterPatch
+end
```

Comment:
This is of course super ugly, but I didn't know how to only patch this when postgresql is in use

Reply:
For this i think we could require only when loading the postgresql schema adapter.
That being said, shouldn't we at this point simply make use of the active record reset_schema_name method when switching tenant?

Reply:
The problem with `reset_sequence_name` is twofold:

1. It goes to the db to find it https://github.com/rails/rails/blob/ecbd92cf48c741f784c1297041d368c8eb14e9bf/activerecord/lib/active_record/model_schema.rb#L357-L360.
2. Modifying the sequence name is not hread safe: https://github.com/rails-on-services/apartment/issues/161


Thus I believe the safest fix is to actually remove the schema from the sequence name